addons/l10n_id_efaktur_coretax/models/account_move_line.py

```diff
@@ -0,0 +1,41 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    def _l10n_id_coretax_build_invoice_line_vals(self, vals):
+        """ Fill in the vals['lines'] with some information regarding each invoice line"""
+
+        self.ensure_one()
+        idr = self.env.ref('base.IDR')
+
+        # initialize
+        if not vals.get('lines'):
+            vals['lines'] = []
+
+        product = self.product_id
+
+        line_val = {
+            "Opt": "B" if product.detailed_type == "service" else "A",  # A: goods, B: service
+            "Code": product.l10n_id_product_code.code or self.env.ref('l10n_id_efaktur_coretax.product_code_000000_goods').code,
+            "Name": product.name,
+            "Unit": self.product_uom_id.l10n_id_uom_code.code,
+            "Price": idr.round(self.price_unit),
+            "Qty": self.quantity,
+            "TotalDiscount": idr.round(self.discount * self.quantity * self.price_unit / 100),
+            "TaxBase": idr.round(self.price_subtotal),  # DPP
+            "VATRate": 12,
```

Comment:
What about Non-taxable Goods & Services?
![image](https://github.com/user-attachments/assets/24391930-9d33-4356-a543-6b829785d4e5)

Reply:
If it's non-taxable, the line should not be declared in E-Faktur I believe. In this case, I think we will need to do filtering on the invoice_line_ids 
i.e.
```
for line in move.invoice_line_ids.filtered(lambda l: l.tax_ids):
                line._l10n_id_coretax_build_invoice_line_vals(vals)
```